neuro_san/client/direct_agent_session_factory.py

```diff
@@ -73,7 +73,7 @@ def create_session(self, agent_name: str, use_direct: bool = False,
         llm_factory.load()
         toolbox_factory.load()
 
-        factory = ExternalAgentSessionFactory(use_direct=use_direct)
+        factory = ExternalAgentSessionFactory(use_direct=use_direct, network_storage=self.network_storage)
```

Comment:
This is the only place where we pass the AgentNetworkStorage into the ExternalAgentSessionFactory.
Other places are in the service and the always pass use_direct=False, whose code paths will never user the network storage.